js/modules/k6/ws/ws.go

```diff
@@ -292,7 +292,8 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 				Tags:   socket.sampleTags,
 				Value:  1,
 			})
-			socket.handleEvent("message", rt.ToValue(string(readData)))
+			ab := rt.NewArrayBuffer(readData)
+			socket.handleEvent("message", rt.ToValue(string(readData)), rt.ToValue(&ab))
```

Comment:
Would this be prohibitively expensive memory wise?

I experimented with overriding the first argument to be a `String` object with an additional `buffer` property, but that doesn't expose the string primitive properties (`.length`, etc.) so it would be a breaking change.

Passing it as a second argument is backwards compatible, and I'm expecting it to be garbage collected if it's not used, but haven't run any benchmarks. One optimization we could do is to only pass it if the `message` handler function expects a second argument, but it would be awkward to do and would require additional refactoring.

Reply:
I think it will be better to just have `binaryMessage` event or something like that instead of ... this ;) 

This is not how it's done in other tools but given the current way it is done in k6 I think this is better and then we can fix it properly in the next implementation when it is more async

Reply:
Sure, see e44fb6d2.

But I also think this is a rough bandaid that adds more work for users once we move to a better implementation and deprecate it.